File: SHIPBOB_TO_GORGIAS/typescript/test/parity.test.ts
// Not in the Python suite: the cron and webhook flows resolve "the same customer" via the
// same GorgiasAPI helpers (customerKey/validEmail), but Python's separate test files never
// cross-check that a matching recipient produces the same key from both an order and a
// shipment-event fixture. Guards against that drifting.

import assert from 'node:assert/strict'
import { test } from 'node:test'

import { GorgiasAPI } from '../src/gorgias.js'
import { GORGIAS_SECRETS, makeOrder, makePandium, makeShipmentEvent } from './helpers.js'

test('customerKey resolves the same key for a matching recipient across an order and a shipment event', () => {
    const api = new GorgiasAPI(makePandium({ secrets: GORGIAS_SECRETS }))

    const order = makeOrder(1, '2026-07-01T00:00:00Z', { email: 'jane@example.com' })
    const event = makeShipmentEvent(1, 'Delivered', 'jane@example.com')
    assert.equal(api.customerKey(order), api.customerKey(event))
})

File: shipbob_to_gorgias/ruby/test/parity_test.rb
require_relative 'test_helper'
require_relative 'fakes'

# Guards against the cron and webhook flows drifting on customer-key derivation, since both
# call GorgiasAPI#customer_key against differently-shaped payloads (an order vs. a shipment
# event) that happen to share the same recipient.* structure.
class ParityTest < Minitest::Test
  def test_customer_key_matches_across_order_and_shipment_event_for_the_same_recipient
    api = Sb2Gorgias::GorgiasAPI.new(make_pandium(secrets: GORGIAS_SECRETS))
    order = make_order(1, '2026-07-01T00:00:00', email: 'jane@example.com')
    event = make_shipment_event(1, 'Delivered', 'jane@example.com')

    assert_equal api.customer_key(order), api.customer_key(event)
  end

  def test_customer_key_matches_for_synthetic_key_too
    api = Sb2Gorgias::GorgiasAPI.new(make_pandium(secrets: GORGIAS_SECRETS))
    recipient = { 'name' => 'Buyer', 'address' => { 'address1' => '1 Main St', 'city' => 'NY', 'country' => 'US' } }
    order = { 'recipient' => recipient }
    event = { 'recipient' => recipient }

    assert_equal 'Buyer 1 Main St NY US', api.customer_key(order)
    assert_equal api.customer_key(order), api.customer_key(event)
  end
end
